### PR TITLE
[instrument_builder] ADDED: field warnings for maximum length in Instrument Builder module

### LINKED ISSUE #7582: [InstrBuilder] check and warn if fieldname is too long (including `_status` fields)
Users of the Instrument Builder should be warned if they are entering a field that will generate a fieldname too long for Mysql.  Better to warn them in the Instrument Builder than to have them encounter an error sourcing the table. 
(Warn the user in the front-end of the instrument builder, and ask them to shorten it.)
Particularly for field types for which a status field is added by the instrument builder to the linst file -- e.g. Textarea (commentbox), text, numeric, BasicDate(?), etc., since this adds an extra 7 characters (`_status`) to the fieldname.

This is likely a beginner-friendly task.

### PR DESCRIPTION
**Pull request by satvik-tha-god**
…t Builder

## Brief summary of changes
* Added field warnings for length for `_status` fields: 
* Added field warnings for length for all the fields
* Minor spelling mistake fix

#### Testing instructions (if applicable)
1. Go to /instrument_builder/#Build 
2. Check for warnings for input length greater than 57 for the following fields by clicking Add Row 
    1. Data entry->Textbox->Question Name
    2. Data entry->Textarea->Question Name
    3. Data entry->Date->Question Name
    4. Data entry->Numeric->Question Name
3. Check for warnings for input length greater than 64 for the rest of the Question Name and Question Text fields by clicking Add Row

#### Link(s) to related issue(s) 

* Resolves #7582

### CODE REVIEW

**Comment by christinerogers** [comment]
Great stuff, Satvik!  It would be great to have an idea of how the warnings will appear to the end user - could you kindly post a screenshot in this thread, showing the error message in context?  Thanks, that would be really helpful in speeding this through the review process.

**Comment by satvik-tha-god** [comment]
> Great stuff, Satvik! It would be great to have an idea of how the warnings will appear to the end user - could you kindly post a screenshot in this thread, showing the error message in context? Thanks, that would be really helpful in speeding this through the review process.

@christinerogers  Thanks!  Here are the screenshots
Screenshot from 2022-06-14 01-50-55
Screenshot from 2022-06-14 01-50-39

File: modules/instrument_builder/jsx/react.questions.js [hunk 1]
```diff
@@ -996,12 +996,80 @@ class AddElement extends Component {
       });
     }
 
+    if(questionName.length > 64 && selected !== 'textbox' && selected !== 'textarea' &&
+      selected !== 'date' && selected !== 'numeric') {
+      // Error, question name is needed for the desired type. Set the element
+      // error flag for the questionName with message. Set the hasError flag
+      let temp = (this.state.error) ? this.state.error : {};
+      temp.questionName = 'Field length should be strictly less than 65 characters';
```
Comment (christinerogers) [inline modules/instrument_builder/jsx/react.questions.js:1004]:
```suggestion
      temp.questionName = 'Please shorten to 64 characters maximum';
```

File: modules/instrument_builder/jsx/react.questions.js [hunk 2]
```diff
@@ -996,12 +996,80 @@ class AddElement extends Component {
...
+      // An error is present, return
+      return;
+    }
+
+    if(questionName.length > 57 && (selected === 'textbox' || selected === 'textarea' || 
+      selected === 'date' || selected === 'numeric')) {
+      // Error, question name is needed for the desired type. Set the element
+      // error flag for the questionName with message. Set the hasError flag
+      let temp = (this.state.error) ? this.state.error : {};
+      temp.questionName = 'Field length should be strictly less than 58 characters';
```
Comment (christinerogers) [inline modules/instrument_builder/jsx/react.questions.js:1049]:
```suggestion
      temp.questionName = 'Please shorten to 57 characters maximum';
```
@satvik-tha-god - please check these carefully as I translated your `less than 58` to `57 maximum` - is this accurate?

**Review by christinerogers — CHANGES_REQUESTED** [review]
Hi @satvik-tha-god thanks for this PR! 
I've suggested edits for the error messages to let the user know directly what they should do (shorten it). Thanks Satvik for checking over the numbers I included. 
@racostas - over to you to check if this satisfies the issue / all cases etc.

**Comment by racostas** [comment]
Hi @satvik-tha-god,
Great work !!
Please address @christinerogers comments.
The unit test are falling due to code standard nonconformity. You will be able to identify the error entering in the failed test and searching for the _**error**_ keyword in the output. For this case are simple ones. I attach a screenshot for reference.

image

**Comment by racostas** [comment]
Not sure why but it seems some of the test got cancelled. 

image

Please relaunch the jobs.

**Comment by racostas** [comment]
failing a the test suit, nevertheless I don't see direct relation avec this PR itself. What do you think @driusan ?

**Review by racostas — CHANGES_REQUESTED** [review]
Thank you @satvik-tha-god for those fixes. 
I think the only thing missing is to update the `modules/instrument_builder/test/TestPlan.md` accordingly. Could you please take a look into it when you get a chance ?
I'm cc @christinerogers as she might want to take a look as well since she identified the original issue.

**Comment by driusan** [comment]
I think we can merge this and then send another PR to fix the test plan.

**Pull request merged**